pkg/cmd/run/view/view.go

```diff
@@ -477,7 +478,8 @@ func promptForJob(cs *iostreams.ColorScheme, jobs []shared.Job) (*shared.Job, er
 func attachRunLog(rlz *zip.ReadCloser, jobs []shared.Job) {
 	for i, job := range jobs {
 		for j, step := range job.Steps {
-			filename := fmt.Sprintf("%s/%d_%s.txt", job.Name, step.Number, step.Name)
+			filename := fmt.Sprintf("%s/%d_%s.txt", job.Name, step.Number,
+				strings.ReplaceAll(step.Name, "/", ""))
```

Comment:
This looks good as a fix! I have a couple of additional asks:

1. Could the step-to-filename–generating logic be extracted into a separate method?
2. Could we have some tests exercising that method?
3. There are possibly some characters other than `/` that don't translate well to filenames (for example: the `:` character). @vilmibm do you think it would be possible to take a look at how our platform generates filenames and match that logic in our codebase? Or, could we somehow avoid guessing filename names altogether?

Reply:
(1) is done. (2) I don't speak go, so you will have to guide me through the testing framework. Or feel free add the tests yourself. (3) Slashes in particular are a quite serious bug because any step of the form `uses: some-org/some-action@v1` will fail unless a name is specified. For example, `run view --log` is useless for reviewing test failures on virtually every Julia lang repo because of this issue. So, my preference is that this is merged and tagged quickly.

Reply:
@mislav @adamslc I think we can avoid special character handling all together by just matching on job name and step number. The job name only allows `-`, `_`, and alphanumeric characters and the step number is an integer so we shouldn't have to recreate the filename generating logic that actions uses and what characters it strips out. I wrote some code that changes this logic and adds tests that I will commit to this PR if that is okay with everyone?

Reply:
@adamslc No worries; thanks for taking us this far.
@samcoe Looks great! Thanks for devising the new approach